synapse/handlers/room_list.py

```diff
@@ -238,9 +238,12 @@ def handle_room(room_id):
                 addition *= -1
 
             try:
-                new_limit = sorted_rooms.index(last_room_id) + addition
-                if new_limit >= len(sorted_rooms):
+                if sorted_rooms.index(last_room_id) == len(sorted_rooms):
```

Comment:
Yes, the index will always be strictly less than len.

A comment as to why you're choosing to stop then might be equally helpful. :)